docs/api.rst

```diff
@@ -259,6 +268,65 @@ defaults in the future might change.
 Inside the templates the behaviour can be temporarily changed by using
 the `autoescape` block (see :ref:`autoescape-overrides`).
 
+Note that :func:`~jinja2.select_autoescape` offers also a parameter
+`special_extensions` that allows you to define a custom escape function,
+i.e. if you handling LaTeX files
+(see description of :func:`~jinja.select_autoescape` above for example).
+
+To write your own custom escape function simply return it for the
+wanted template extension. For instance if you are fan of peace in
+the world::
+
+    def escape_to_peace(s):
+        """
+        Your custom escape function. You only have to take care
+        that your escaping is done properly, everything else like
+        preventing multiple escapes and marking the string as safe
+        is done by Jinja itself.
+        """
+        s.replace("war", "peace")
+
+    env = Environment(
+        autoescape=select_autoescape(special_extensions={".world": escape_to_peace}),
+        loader=PackageLoader("mypackage"),
+    )
+
+
+Note that for `.world` files the `{{ var|e }}` and `{{ var | escape }}`
```

Comment:
I think all this should use double backticks (assuming you want the equivalent to `single-backtick` highlighting on github)